djangomako/backends.py

```diff
@@ -164,7 +165,37 @@ def render(self, context=None, request=None):
             context['csrf_input'] = csrf_input_lazy(request)
             context['csrf_token'] = csrf_token_lazy(request)
 
-        return self.template.render(**context)
+        try:
+            return self.template.render(**context)
+        except Exception as e:
+            tb = RichTraceback()
+
+            lines = tb.source
+            if not lines:
+                # There's no templates lines then raise
+                raise e
+
+            lines = lines.split('\n')
+            line = tb.lineno
+            top = max(0, line - 4)
+            bottom = min(len(lines), line + 5)
+            source_lines = [(i+1, lines[i] + '\n') for i in range(top, bottom)]
+
+            e.template_debug = {
+                'name': tb.traceback[len(tb.traceback) - 1][0],
+                'message': '{}: {}'.format(tb.errorname, tb.message),
```

Comment:
Given that error_details are given in: ` traceback.records[5]`, let:
```PYTHON
'message': '{}: {}'.format(traceback.errorname, function_name)
# Where
function_name = error_details[3]
```

Reply:
Unfortunately this is not what's targeted, this `function_name` will return which mako function is raising, like for example `__M_writer(str(some_func_call()))`, but what's targeted here is the error message like `'Undefined' object is not callable`

Reply:
The idea is to have something like this as result : 

```
TypeError: 'Undefined' object is not callable
```